Source/Utils/SpeciesUtils.cpp

```diff
@@ -58,6 +50,22 @@ namespace SpeciesUtils {
             species_name + "'."
         );
 
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            mass_is_specified ||
+            species_is_specified ||
+            (injection_style == "external_file"),
```

Comment:
In the external file logic, is the particle mass truly required or optional?

If it is the latter case, you need to check there as well.